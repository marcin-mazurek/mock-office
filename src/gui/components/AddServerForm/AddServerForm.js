import React from 'react';
import PropTypes from 'prop-types';
import Select from 'react-select';
import Switch from 'rc-switch';
import { Field } from 'redux-form/immutable';
import plusIcon from '../../../../assets/icons_white_add.svg';

const ServerTypeField = field =>
  <Select
    name="server-type"
    value={field.input.value}
    onChange={option => field.input.onChange(option.value)}
    searchable={false}
    clearable={false}
    className="form-field__select"
    options={[
      { value: 'http', label: 'HTTP' },
      { value: 'ws', label: 'Web Socket' }
    ]}
  />;

const secureField = field =>
  <Switch
    checked={field.input.value}
    onChange={value => field.input.onChange(value)}
    checkedChildren="Yes"
    unCheckedChildren="No"
    className="form-field__switch"
  />;

const AddServerForm = ({ handleSubmit }) =>
  <form className="form add-server-form" onSubmit={handleSubmit}>
    <div className="form-row">
      <div className="form__field">
        <label className="form-field__label" htmlFor="name">
          Name:
        </label>
        <Field
          component="input"
          type="text"
          className="form-field__input"
          name="name"
          placeholder={'Awesome server'}
        />
      </div>
    </div>
    <div className="form-row">
      <div className="form__col">
        <div className="form__field">
          <label className="form-field__label" htmlFor="port">
            Port:
          </label>
          <Field
            component="input"
            className="form-field__input"
            name="port"
            type="number"
          />
        </div>
      </div>
      <div className="form__col">
        <div className="form__field">
          <label className="form-field__label" htmlFor="type">
            Type:
          </label>
          <Field component={ServerTypeField} name="type" />
        </div>
      </div>
      <div className="form__col form__col--small">
        <div className="form__field">
          <label className="form-field__label" htmlFor="secure">
            Secure:
          </label>
          <Field name="secure" component={secureField} />
        </div>
      </div>
    </div>
    <div className="form-row">
      <button
        className="button form__button"
        type="submit"
      >
        <img className="form__submit-icon" src={plusIcon} alt="" />Create
      </button>
    </div>
  </form>;

AddServerForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default AddServerForm;
