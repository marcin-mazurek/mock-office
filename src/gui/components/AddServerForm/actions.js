export const SUBMIT_BUTTON_CLICKED = 'addServerForm/SUBMIT_BUTTON_CLICKED';
export const submitButtonClickAction = values => ({
  type: SUBMIT_BUTTON_CLICKED,
  values
});
