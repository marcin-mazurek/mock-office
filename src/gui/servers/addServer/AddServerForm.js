import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Select from 'react-select';
import Switch from 'rc-switch';
import { Field, reduxForm } from 'redux-form/immutable';
import plusIcon from '../../assets/icons_white_add.svg';

export const SUBMIT_BUTTON_CLICKED = 'ddServerForm/SUBMIT_BUTTON_CLICKED';
export const submitButtonClickAction = values => ({
  type: SUBMIT_BUTTON_CLICKED,
  values
});

const INITIAL_VALUES = {
  name: 'Awesome server',
  port: 3000,
  type: 'http',
  secure: false
};

const FORM_NAME = 'addServer';

const ServerTypeField = field =>
  <Select
    name="server-type"
    value={field.input.value}
    onChange={option => field.input.onChange(option.value)}
    searchable={false}
    clearable={false}
    className="form-field__select"
    options={[
      { value: 'http', label: 'HTTP' },
      { value: 'ws', label: 'Web Socket' }
    ]}
  />;

const secureField = field =>
  <Switch
    checked={field.input.value}
    onChange={value => field.input.onChange(value)}
    checkedChildren="Yes"
    unCheckedChildren="No"
    className="form-field__switch"
  />;

export const AddServerForm = ({ handleSubmit }) =>
  <form className="form" onSubmit={handleSubmit}>
    <div className="form-row">
      <div className="form__field">
        <label className="form-field__label" htmlFor="name">
          Name:
        </label>
        <Field
          component="input"
          type="text"
          className="form-field__input"
          name="name"
          placeholder={'Awesome server'}
        />
      </div>
    </div>
    <div className="form-row">
      <div className="form__col">
        <div className="form__field">
          <label className="form-field__label" htmlFor="port">
            Port:
          </label>
          <Field
            component="input"
            className="form-field__input"
            name="port"
            type="number"
          />
        </div>
      </div>
      <div className="form__col">
        <div className="form__field">
          <label className="form-field__label" htmlFor="type">
            Type:
          </label>
          <Field component={ServerTypeField} name="type" />
        </div>
      </div>
      <div className="form__col form__col--small">
        <div className="form__field">
          <label className="form-field__label" htmlFor="secure">
            Secure:
          </label>
          <Field name="secure" component={secureField} />
        </div>
      </div>
    </div>
    <div className="form-row">
      <button
        className="button form__button"
        type="submit"
      >
        <img className="form__submit-icon" src={plusIcon} alt="" />Create
      </button>
    </div>
  </form>;

AddServerForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  onSubmit: submitButtonClickAction
};

export default connect(null, mapDispatchToProps)(
  reduxForm({
    form: FORM_NAME,
    initialValues: INITIAL_VALUES
  })(AddServerForm)
);
