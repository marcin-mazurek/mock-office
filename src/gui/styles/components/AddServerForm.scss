.add-server-form {
  padding: 15px;
  max-width: 600px;
  width: 100%;
}

.add-server-form-row {
  display: flex;
  margin: 40px auto;
  width: 100%;
}

.add-server-form__col {
  flex: 2;
  margin-left: 20px;
  margin-right: 20px;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
}

.add-server-form__col--small {
  flex: 1;
}

.add-server-form__header {
  font-size: 24px;
  text-align: center;
  font-weight: 500;
  width: 100%;
  margin-bottom: 40px;
}

.add-server-form__header-icon {
  display: block;
  margin: 12px auto;
}

.add-server-form__field {
  width: 100%;
}

.add-server-form__label {
  display: block;
  font-size: 13px;
  color: #999;
  margin-bottom: 3px;
}

.add-server-form__submit-button {
  cursor: pointer;
  font-size: 16px;
  background-color: #23D1A9;
  border-radius: 100px;
  padding: 12px 27px;
  font-weight: 500;
  color: #FFF;
  margin: 60px auto 0;
  display: block;
}

.add-server-form__submit-icon {
  display: inline-block;
  height: 12px;
  margin-right: 5px;
}
