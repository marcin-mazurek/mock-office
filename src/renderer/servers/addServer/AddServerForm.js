import React from 'react';
import { connect } from 'react-redux';
import { Textfield, Button } from 'react-mdl';
import init from './actions';

export class AddServerForm extends React.Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
    this.handleNameChange = this.handleNameChange.bind(this);
    this.handleChangePort = this.handleChangePort.bind(this);
    this.handleTypeChange = this.handleTypeChange.bind(this);
    this.handleSecureChange = this.handleSecureChange.bind(this);
    this.handleKeyChange = this.handleKeyChange.bind(this);
    this.handleCertChange = this.handleCertChange.bind(this);

    this.state = {
      name: 'Server name',
      port: 3000,
      type: 'http',
      isSecure: 'no',
      keyPath: '',
      cartPath: ''
    };
  }

  handleNameChange(e) {
    this.setState({
      name: e.currentTarget.value
    });
  }

  handleChangePort(e) {
    this.setState({
      port: parseInt(e.currentTarget.value, 10)
    });
  }

  handleTypeChange(e) {
    this.setState({
      type: e.currentTarget.value
    });
  }

  handleSecureChange(e) {
    this.setState({
      isSecure: e.currentTarget.value
    });
  }

  handleKeyChange(e) {
    this.setState({
      keyPath: e.currentTarget.files[0].path
    });
  }

  handleCertChange(e) {
    this.setState({
      cartPath: e.currentTarget.files[0].path
    });
  }

  submit(event) {
    event.preventDefault();
    this.props.add(
      this.state.name,
      this.state.port,
      this.state.type,
      this.state.isSecure === 'yes',
      this.state.keyPath,
      this.state.cartPath
    );
  }

  render() {
    return (
      <div className="add-server-form">
        <form action="#">
          <div className="add-server-form__field">
            <Textfield
              floatingLabel
              onChange={this.handleNameChange}
              label="Name"
              value={this.state.name}
            />
          </div>
          <div className="add-server-form__field">
            <Textfield
              floatingLabel
              onChange={this.handleChangePort}
              label="Port"
              value={this.state.port}
            />
          </div>
          <div className="add-server-form__field">
            <label
              className=""
              htmlFor="type"
            >
              Type: {' '}
            </label>
            <select
              id="type"
              name="type"
              value={this.state.type}
              onChange={this.handleTypeChange}
            >
              <option value="http">http</option>
              <option value="ws">ws</option>
            </select>
          </div>
          <div className="add-server-form__field">
            <label htmlFor="secure">
              Secure: {' '}
            </label>
            <select
              name="secure"
              id="secure"
              value={this.state.isSecure}
              onChange={this.handleSecureChange}
            >
              <option value="yes">yes</option>
              <option value="no">no</option>
            </select>
          </div>
          {
            this.state.isSecure === 'yes'
              ? (
                <div>
                  <div className="add-server-form__field">
                    <label
                      className=""
                      htmlFor="type"
                    >
                      Key: {' '}
                    </label>
                    <input type="file" onChange={this.handleKeyChange} />
                  </div>
                  <div className="add-server-form__field">
                    <label
                      className=""
                      htmlFor="type"
                    >
                      Certificate: {' '}
                    </label>
                    <input type="file" onChange={this.handleCertChange} />
                  </div>
                </div>
              )
              : null
          }
          <Button
            raised
            colored
            ripple
            onClick={this.submit}
          >
            Create
          </Button>
        </form>
      </div>
    );
  }
}

AddServerForm.propTypes = {
  add: React.PropTypes.func.isRequired
};

const mapDispatchToProps = {
  add: init
};

export default connect(null, mapDispatchToProps)(AddServerForm);
