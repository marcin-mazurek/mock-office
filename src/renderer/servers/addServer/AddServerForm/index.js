import React from 'react';
import { connect } from 'react-redux';
import { Textfield, Button } from 'react-mdl';
import init from '../actions';

class AddServerForm extends React.Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
    this.handleNameChange = this.handleNameChange.bind(this);
    this.handleChangePort = this.handleChangePort.bind(this);
    this.handleTypeChange = this.handleTypeChange.bind(this);

    this.state = {
      name: 'Server name',
      port: 3000,
      type: 'http'
    };
  }

  handleNameChange(e) {
    this.setState({
      name: e.currentTarget.value
    });
  }

  handleChangePort(e) {
    this.setState({
      port: parseInt(e.currentTarget.value, 10)
    });
  }

  handleTypeChange(e) {
    this.setState({
      type: e.currentTarget.value
    });
  }

  submit(event) {
    event.preventDefault();
    this.props.add(this.state.name, this.state.port, this.state.type);
  }

  render() {
    return (
      <div className="add-server-form">
        <form action="#">
          <div className="add-server-form__field">
            <Textfield
              floatingLabel
              onChange={this.handleNameChange}
              label="Name"
              value={this.state.name}
            />
          </div>
          <div className="add-server-form__field">
            <Textfield
              floatingLabel
              onChange={this.handleChangePort}
              label="Port"
              value={this.state.port}
            />
          </div>
          <div className="add-server-form__field">
            <label
              className=""
              htmlFor="type"
            >
              Type: {' '}
            </label>
            <select
              id="type"
              name="type"
              value={this.state.type}
              onChange={this.handleTypeChange}
            >
              <option value="http">http</option>
              <option value="ws">ws</option>
            </select>
          </div>
          <Button
            raised
            colored
            ripple
            onClick={this.submit}
          >
            Create
          </Button>
        </form>
      </div>
    );
  }
}

AddServerForm.propTypes = {
  add: React.PropTypes.func.isRequired
};

const mapDispatchToProps = {
  add: init
};

export default connect(null, mapDispatchToProps)(AddServerForm);
