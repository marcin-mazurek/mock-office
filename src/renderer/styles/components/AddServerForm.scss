.add-server-form {
  padding: 15px;
}

.add-server-form__field {
  display: block;
  margin: 10px 0;
}
