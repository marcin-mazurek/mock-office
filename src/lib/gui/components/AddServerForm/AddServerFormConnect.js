import { connect } from 'react-redux';
import { reduxForm } from 'redux-form/immutable';
import AddServerForm from './AddServerForm';
import { submitButtonClickAction } from './actions';

const FORM_NAME = 'addServer';
const INITIAL_VALUES = {
  name: 'Awesome server',
  port: 3000,
  type: 'http',
  secure: false
};
const mapDispatchToProps = {
  onSubmit: submitButtonClickAction
};

export default connect(null, mapDispatchToProps)(
  reduxForm({
    form: FORM_NAME,
    initialValues: INITIAL_VALUES
  })(AddServerForm)
);
