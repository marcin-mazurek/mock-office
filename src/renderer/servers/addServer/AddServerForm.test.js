import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AddServerForm } from './AddServerForm';

test('AddServerForm should render', () => {
  const wrapper = shallow(<AddServerForm add={() => {}} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
