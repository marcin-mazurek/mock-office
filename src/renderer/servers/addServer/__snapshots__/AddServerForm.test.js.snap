exports[`test AddServerForm should render 1`] = `
<div
  className="add-server-form">
  <form
    action="#">
    <div
      className="add-server-form__field">
      <Textfield
        floatingLabel={true}
        label="Name"
        onChange={[Function]}
        value="Server name" />
    </div>
    <div
      className="add-server-form__field">
      <Textfield
        floatingLabel={true}
        label="Port"
        onChange={[Function]}
        value={3000} />
    </div>
    <div
      className="add-server-form__field">
      <label
        className=""
        htmlFor="type">
        Type: 
         
      </label>
      <select
        id="type"
        name="type"
        onChange={[Function]}
        value="http">
        <option
          value="http">
          http
        </option>
        <option
          value="ws">
          ws
        </option>
      </select>
    </div>
    <div
      className="add-server-form__field">
      <label
        htmlFor="secure">
        Secure: 
         
      </label>
      <select
        id="secure"
        name="secure"
        onChange={[Function]}
        value="no">
        <option
          value="yes">
          yes
        </option>
        <option
          value="no">
          no
        </option>
      </select>
    </div>
    <Button
      colored={true}
      onClick={[Function]}
      raised={true}
      ripple={true}>
      Create
    </Button>
  </form>
</div>
`;
